source/extensions/filters/network/redis_proxy/proxy_filter.h

```diff
@@ -139,6 +139,8 @@ class ProxyFilter : public Network::ReadFilter,
     Common::Redis::Client::Transaction& transaction() override { return parent_.transaction(); }
 
     ProxyFilter& parent_;
+    // This value is set when the request is on hold, waiting for an external auth response.
+    Common::Redis::RespValuePtr pending_value_;
```

Comment:
nit: I would call this `pending_request_value_` or just `pending_request_`. Technically I think you could merge `pending_value_` and `pending_response_` as I don't think they can ever be set at the same time, but not sure that is worth it.

Reply:
The problem with merging is that there is a condition where if `pending_response_` is set, the pending responses get flushed, when `onResponse` is called, and the request is removed from queue.
Renaming to `pending_request_value_` as suggested 😄